### PR TITLE
fix(#16762): should use cachedOptions when filtered options don't cange

### LINKED ISSUE #16762: Cascader: User has to press down arrow key twice to select the first option while searching
- [x] I have searched the [issues](https://github.com/ant-design/ant-design/issues) of this repository and believe that this is not a duplicate.

### Reproduction link
[Edit on CodeSandbox](https://codesandbox.io/s/yq9pj460rx)

### Steps to reproduce
When searching in the cascader, to select item using key board user has to press the down arrow key twice to focus the list item.

Is there any option to select the first item by default when searching ?
Something like defaultActiveFirstOption ?

https://codesandbox.io/s/yq9pj460rx

### What is expected?
After entering search if user click down arrow the first option should select 

### What is actually happening?
After entering search  user has to click down arrow twice to select the first option 

| Environment | Info |
|---|---|
| antd | 3.18.2 |
| React | any |
| System | Any |
| Browser | Any |

### PR DESCRIPTION
**Pull request by kerm1it**
### 🤔 This is a ...

- [ ] New feature
- [x] Bug fix
- [ ] Site / document update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Other (about what?)

### 🔗 Related issue link

close #16762

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English | Fix Cascader search need to press down arrow twice to select item(#16762)       |
| 🇨🇳 Chinese |  修复 Cascader 搜索时需要按两次向下箭头才能选中问题(#16762)         |

### ☑️ Self Check before Merge

⚠️ Please check all items below before review. ⚠️ 

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

### CODE REVIEW

**Comment by Kermit-Xuan** [comment]
The problem is that the search options always generates new, even when the options has not changed, causing component always re-render.

File: components/cascader/index.tsx [hunk 1]
```diff
@@ -3,6 +3,7 @@ import RcCascader from 'rc-cascader';
 import arrayTreeFilter from 'array-tree-filter';
 import classNames from 'classnames';
 import omit from 'omit.js';
+import isEqual from 'lodash/isEqual';
```
Comment (afc163) [inline components/cascader/index.tsx]: Try `shallowequal`

Comment (Kermit-Xuan) [reply components/cascader/index.tsx]: I can't find `shallowequal` in package.json, need I set up it?

Comment (afc163) [reply components/cascader/index.tsx]: lodash isEqual is ok then.

**Comment by afc163** [comment]
image

Could you fill text cov?

**Comment by Kermit-Xuan** [comment]
> image
> 
> Could you fill text cov?

I don't know that how to do it?

**Comment by Kermit-Xuan** [comment]
> image
> 
> Could you fill text cov?

Sorry, I don't know how to do it.

**Comment by Kermit-Xuan** [comment]
In progress?

**Comment by afc163** [comment]
https://codecov.io/gh/ant-design/ant-design/pull/22216/diff?src=pr&el=tree#diff-Y29tcG9uZW50cy9jYXNjYWRlci9pbmRleC50c3g=

This line is not be covered.

**Comment by Kermit-Xuan** [comment]
> [codecov.io/gh/ant-design/ant-design/pull/22216/diff?src=pr&el=tree#diff-Y29tcG9uZW50cy9jYXNjYWRlci9pbmRleC50c3g=](https://codecov.io/gh/ant-design/ant-design/pull/22216/diff?src=pr&el=tree#diff-Y29tcG9uZW50cy9jYXNjYWRlci9pbmRleC50c3g=)
> 
> This line is not be covered.

I try it in my local, this line is be covered.

WX20200319-134111@2x

**Comment by afc163** [comment]
Could you fill the changelog part in which we should tell our users?

image

**Comment by Kermit-Xuan** [comment]
> Could you fill the changelog part in which we should tell our users?
> 
> image

Done.

**Pull request merged**